router/src/server/http.rs

```diff
@@ -385,6 +404,20 @@ where
         let namespace = org_and_bucket_to_namespace(&write_info.org, &write_info.bucket)
             .map_err(OrgBucketError::MappingFail)?;
 
+        if let Some(authz) = &self.authz {
+            if let Some(tok) = v2_token(&req) {
+                let perms = vec![Permission::ResourceAction(
+                    Resource::Namespace(namespace.to_string()),
+                    Action::Write,
+                )];
+                if authz.allow(&tok, &perms).await?.is_empty() {
+                    return Err(Error::Forbidden);
+                }
+            } else {
+                return Err(Error::Unauthenticated);
+            }
```

Comment:
This will result in one less allocation, and is a little easier to read:


```suggestion
            let token = v2_token(&req).ok_or(Error::Unauthenticated)?;

            let perms = [Permission::ResourceAction(
                Resource::Namespace(namespace.to_string()),
                Action::Write,
            )];
            if authz.allow(&token, &perms).await?.is_empty() {
                return Err(Error::Forbidden);
            }
```

Combined with the above, this block could be even cleaner!